pkg/asset/installconfig/azure/client.go

```diff
@@ -211,3 +212,60 @@ func (c *Client) ListResourceIDsByGroup(ctx context.Context, groupName string) (
 	}
 	return res, nil
 }
+
+// GetVirtualMachineSku retrieves the restource SKU of a specified virtual machine in the given region.
+func (c *Client) GetVirtualMachineSku(ctx context.Context, name, region string) (azsku.ResourceSku, error) {
+	results, err := c.GetResourceSkus(ctx, "virtualMachines", region)
+	if err != nil {
+		return azsku.ResourceSku{}, err
+	}
+
+	for _, sku := range results {
+		if strings.EqualFold(name, *sku.Name) {
+			return sku, nil
+		}
+	}
+
+	return azsku.ResourceSku{}, errors.Errorf("SKU not found for specified subscription in region %s", region)
+
+}
+
+// GetResourceSkus retrieves the resource SKUs of a specified type for a given region.
+func (c *Client) GetResourceSkus(ctx context.Context, resourceType, region string) ([]azsku.ResourceSku, error) {
+	client := azsku.NewResourceSkusClientWithBaseURI(c.ssn.Environment.ResourceManagerEndpoint, c.ssn.Credentials.SubscriptionID)
+	client.Authorizer = c.ssn.Authorizer
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	var results []azsku.ResourceSku
+
+	for page, err := client.List(ctx); page.NotDone(); err = page.NextWithContext(ctx) {
```

Comment:
It is unfortunate that the Azure client does not easily support adding the location filter. In the subscription that I am using, there are 7757 virtual machine SKUs that all need to be fetched to validate the instance type.

Reply:
I totally agree.